src/engine/ruleset/decoders/fortinet/firewall.yml

```diff
@@ -0,0 +1,534 @@
+    # - ~tmp.user: +s_ne/N\/A
+    map:
+      - source.user.name: $~tmp.user
+      - related.user: [$source.user.name]
+  - check:
+    - ~tmp.dstip: +ef_exists
+    map:
+      - destination.ip: $~tmp.dstip
+  - check:
+    - ~tmp.remip: +ef_exists
+    - ~tmp.dstip: +ef_not_exists
+    map:
+      - destination.ip: $~tmp.remip
+  - check:
+    - ~tmp.remport: +ef_exists
+    map:
+      - destination.port: $~tmp.remport
+  - check:
+    - ~tmp.dstport: +ef_exists
+    map:
+      - destination.port: $~tmp.dstport
+  - check:
+    - ~tmp.srcip: +ef_exists
+    map:
+      - source.ip: $~tmp.srcip
+  - check:
+    - ~tmp.locip: +ef_exists
+    - ~tmp.srcip: +ef_not_exists
+    map:
+      - source.ip: $~tmp.locip
+  - map:
+    - related.ip: [$source.ip]
+  - check:
+    - ~tmp.locport: +ef_exists
+    map:
+      - source.port: $~tmp.locport
+  - check:
+    - ~tmp.srcport: +ef_exists
+    map:
+      - source.port: $~tmp.srcport
+  - check:
+    - log.level: error
+    map:
+      - event.outcome: failure
+  - check:
+    - source.ip: +ef_exists
+    map:
+      - ~tmp.ip: +s_ip_version/$source.ip
+  - check:
+    - destination.ip: +ef_exists
+    - source.ip: +ef_not_exists
+    map:
+      - ~tmp.ip: +s_ip_version/$destination.ip
+  - map:
+    - network.type: +s_lo/$~tmp.ip
+
+  # Check-mapping for firewall-event
+  - check:
+    - ~tmp.subtype: user
+    - event.action: +r_match/(?i)logon
+    map:
+      - event.type: [start, user]
+  - check:
+    - ~tmp.subtype: user
+    - event.action: +r_match/(?i)logoff
+    map:
+      - event.type: [end, user]
+  - map:
+    - event.category: [authentication]
+  - check:
+    - ~tmp.subtype: system
+    map:
+      - event.category: [host]
+      - event.type: [info]
+      - event.outcome: +ef_delete
+
+  # Check-mapping for firewall-traffic
+  - check:
+    - ~tmp.subtype: vpn
+    map:
+      - event.category: [network]
+      - event.type: [connection]
+  - check: >-
+        (+s_eq/~tmp.type/traffic) OR (+s_eq/~tmp.type/utm)
+    map:
+      - event.category: [network]
+  - check:
+    - ~tmp.type: traffic
+    map:
+      - event.type: [connection]
+  - check: >-
+        (+s_eq/~tmp.type/traffic) AND ((+s_eq/event.action/block) OR (+s_eq/event.action/blocked) OR
+        (+s_eq/event.action/deny) OR (+s_eq/event.action/close) OR
+        (+s_eq/event.action/server-rst) OR (+s_eq/~tmp.utmaction/block))
+    map:
+      - event.type: +a_append/denied
+  - check:
+    - ~tmp.type: traffic
+    - ~tmp.utmaction: +ef_not_exists
+    - event.action: +s_ne/block
+    - event.action: +s_ne/blocked
+    - event.action: +s_ne/deny
+    - event.action: +s_ne/close
+    - event.action: +s_ne/server-rst
+    map:
+      - event.type: +a_append/allowed
+  - check:
+    - ~tmp.type: traffic
+    - event.action: allow
+    map:
+      - event.type: +a_append/allowed
+  - check:
+    - ~tmp.type: traffic
+    - event.action: +s_eq/start
+    map:
+      - event.type: +a_append/start
```

Comment:
Sometimes `+s_eq` is used and sometimes not, please, if possible, try to always use the same criteria